src/datastore/src/userstore.js

```diff
@@ -33,6 +35,54 @@ class UserStore extends NetworkStore {
   }
 
   /**
+   * Find all users. A query can be optionally provided to return
+   * a subset of all users or omitted to return all users.
+   *
+   * @param {Query} [query] Query used to filter entities.
+   * @param {Object} [options] Options
+   * @param {Boolean} [options.discover] Discover users.
+   * @return {Observable} Observable.
+   */
+  find(query, options = {}) {
+    // If `options.discover`, use
+    // [User Discovery](http://devcenter.kinvey.com/guides/users#lookup)
+    // instead of querying the user namespace directly.
+    if (options.discover === true) {
```

Comment:
@thomasconner Why are we overloading the `user.find` method? On iOS and Xamarin, for example, we have a separate method called `User.lookup()`. Since the REST API used for lookup is different, I'd prefer if we separated the methods. 
Btw, what is `find()` on users supposed to do? If it's a supported operation, we need unit tests to cover the expected result.